File: tracks/a-jsonl-cli/tasks_cli.py
#!/usr/bin/env python3
import json
import sys
from collections import Counter, defaultdict

if len(sys.argv) != 2:
    print("usage: tasks_cli.py <tasks.jsonl>", file=sys.stderr)
    sys.exit(1)

status_counts = Counter()
priority_points = defaultdict(int)
total = 0

with open(sys.argv[1], "r", encoding="utf-8") as f:
    for line in f:
        line = line.strip()
        if not line:
            continue
        task = json.loads(line)
        total += 1
        status_counts[task["status"]] += 1
        priority_points[task["priority"]] += int(task.get("estimate", 0))

print(f"total_tasks={total}")
for status in ["todo", "in_progress", "done"]:
    print(f"status_{status}={status_counts.get(status, 0)}")
for priority in ["high", "medium", "low"]:
    print(f"points_{priority}={priority_points.get(priority, 0)}")

File: tracks/a-jsonl-cli/tasks_cli.ts
#!/usr/bin/env node
import { readFileSync } from 'node:fs';

const input = process.argv[2];
if (!input) {
  console.error('usage: tasks_cli.ts <tasks.jsonl>');
  process.exit(1);
}

const statusCounts = new Map<string, number>();
const priorityPoints = new Map<string, number>();
let total = 0;

for (const rawLine of readFileSync(input, 'utf8').split(/\r?\n/)) {
  const line = rawLine.trim();
  if (!line) continue;
  const task = JSON.parse(line) as {
    status: string;
    priority: string;
    estimate?: number;
  };
  total += 1;
  statusCounts.set(task.status, (statusCounts.get(task.status) ?? 0) + 1);
  priorityPoints.set(task.priority, (priorityPoints.get(task.priority) ?? 0) + (task.estimate ?? 0));
}

console.log(`total_tasks=${total}`);
for (const status of ['todo', 'in_progress', 'done']) {
  console.log(`status_${status}=${statusCounts.get(status) ?? 0}`);
}
for (const priority of ['high', 'medium', 'low']) {
  console.log(`points_${priority}=${priorityPoints.get(priority) ?? 0}`);
}
